clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -505,7 +497,7 @@ public boolean poll(Timer timer, boolean waitForJoinGroup) {
                 }
 
                 // if not wait for join group, we would just use a timer of 0
-                if (!ensureActiveGroup(waitForJoinGroup ? timer : time.timer(0L))) {
+                if (!ensureActiveGroup(waitForJoinGroup ? timer : time.timer(0L), timer)) {
```

Comment:
I don't think we should also pass `timer` when not `waitForJoinGroup` here. I think when not `waitForJoinGroup`, we should not do `onJoinPrepare`, so we should pass `time.timer(0L)`, too. What do you think?